doc/reports/spikes/ras-rm-account-removal.md

```diff
@@ -0,0 +1,55 @@
+
+With obfuscation being a light touch approach to removing the record of an account, the messages that were sent between the ONS and the respondent can remain untouched as secure-message only ever dealt in uuids from party and never names.
+
+This should be done via an endpoint in ras-party and the auth service (whichever one is being used) as opposed to a script.
+
+# Deleting respondent entirely
+
+I'll go service by service that could be affected by the account removal.  Each service will include what needs to be done to remove it, and what impact removing the account could have. The sample, action, collectionexercise services all have the url to the party service in their configuration.  We'll need to double check that a search that returns nothing doesn't break anything because of a deleted account doesn't break anything.
+
+#### ras-party
+From the steps below, it's possible to unpick an account but the order is important because of foreign/composite/primary keys.  Once the order is determined then it should be possible to automate it.  This order looks roughly like:
+
+
+  - Identify which respondent needs to be removed in `partysvc -> respondent`.  Make a note of the id, party_uuid and email address.
+  - In the `party_svc -> business_respondent` table, identify all of the `business_id`s that have a respondent_id that is the same as the id obtained in the previous step.
+
+   - Remove all of the records from `partysvc -> enrolment` where the respondent_id matches the id (from the first step).  There is a composite key of business_id and respondent_id between enrolment and business_respondent tables.
+   - Remove all of the records with the id from `partysvc -> business_respondent` -> respondent_id. The 'id' of the respondant is a foreign key of respondent id in the business_respondant table.
+   - Remove from `partysvc -> respondent` by email address.
+
+#### ras-rm-auth-service and django-oauth2-test
+Doing the above will make it so that the user cannot log in.  To remove the record of the user from the system, you also have to do the following
+ - Remove from `auth -> user` by email address (if using ras-rm-auth-service)
+- Remove from `public -> credentials_oauthuser` by email address (if using django-oauth service)
+
+#### ras-collection-exercise
+This service shouldn't have any issues if an account gets deleted. It goes to the party service for a business id.  If this has been deleted because of the previous step then it will fail.  Failure (from what I can tell) means that ultimately a SEFT survey (or collection instrument) can't be uploaded.  This is probably fine as a business would need a respondent with an account in order to be able to upload one, if there was only 1 and we deleted it, then they can't get to that screen.  If they have multiple respondents with accounts (is this possible?) then there will be a record in the business table for the other respondent so it should succeed.
```

Comment:
Deleting a user's account shouldn't result in failure in a sense - if a business doesn't have an active account then they won't be able to download a SEFT instrument in the first place.